arch/x86/core/fatal.c

```diff
@@ -24,6 +24,37 @@
 
 __weak void _debug_fatal_hook(const NANO_ESF *esf) { ARG_UNUSED(esf); }
 
+#if defined(CONFIG_EXCEPTION_STACK_TRACE)
+struct stack_frame {
+	struct stack_frame *next;
+	u32_t ret_addr;
+	u32_t args;
+};
+
+static void unwind_stack(u32_t base_ptr)
+{
+	struct stack_frame *frame = (struct stack_frame *)base_ptr;
+	int i;
+
+	if (!frame) {
+		printk("NULL frame ptr\n");
+		return;
+	}
+	if (base_ptr % sizeof(base_ptr) != 0) {
+		printk("unaligned frame ptr\n");
+		return;
+	}
+
+	for (i = 0; i < 8; i++) {
+		if (!frame || !frame->ret_addr) {
```

Comment:
I'd move the `if (base_ptr is aligned)` check to see if the pointer is aligned at every iteration.  Should catch some call stack chain corruption as well.